### ISSUE TITLE
Transmission daemon running under custom user trying to load config from wrong path

### ISSUE THREAD

**Issue by bolinocroustibat** [transmission/transmission#1527]
I have transmission daemon 2.94 running as a service on a Debian 10 machine.

I changed the user running transmission daemon from default `debian-transmission` user to another user (in my case, `www-data` Apache user, since I want the downloaded files to be readable by Apache and the opposite, Apache files to be readable by transmission).

To do that, I changed to the user setting from the following config files:

`User=www-data` in `/lib/systemd/system/transmission-daemon.service`
`USER=www-data` in `/etc/init.d/transmission-daemon`
`setuid www-data` and `setgid www-data` in `/etc/init/transmission-daemon.conf`

The transmission JSON config file still reside in `/etc/transmission-daemon/settings.json`. I also made sure both directories `/var/lib/transmission-daemon` and `/etc/transmission-daemon` were owned by the new user `www-data`.

But when I relaunch the service with `sudo systemctl start transmission-daemon` or `sudo service transmission-daemon start`, I get the following in the transmission debug logs:

```
Couldn't create "/var/www/.config": Permission denied (file-posix.c:189)
Couldn't read "/var/www/.config/transmission-daemon/settings.json": No such file or directory (utils.c:238)
Couldn't save temporary file "/var/www/.config/transmission-daemon/settings.json.tmp.jq3qtN": No such file or directory (variant.c:1285)
```
...and other similar errors.

So it would seem transmission-daemon is trying to use the `www-data` user home path and recreate its config directory here.

Why is that and how can I make transmission using the previous default config path? I don't see any transmission configuration depending on the user home path.

**Comment by bolinocroustibat** [comment]
Found the issue. Transmission seems to always use the /.config dir in the user home folder, and this is not configurable. So I just created a /.config dir and moved the /.config content from the previous user home folder to the new one, with the appropriate rights.
The issue can be closed.

**Comment by kushwavez** [comment]
After a very frustrating and long hour FINALLY with the above mentioned daemon user changes Transmission can access my frikin drive and write to it... Thank you very much! 
I know there is a reason why it is running as a separate user, but man running as main (logged in) USER should be DEFAULT.

Searched numerous sites, of course they didn't work. I made chown to debian-transmission, didn't work, then made chmod a+x, didn't work... I don't know why is it so hard to configure these very simple applications on Linux (also I don't know why they can't just make it as a logged in user perm???)... yuck. Always that disgusting permission issue... always.

Anyways thanks again :)

**Comment by bolinocroustibat** [comment]
I agree this is pretty obscure and frustrating.
Also, it seems that it changes user back to default regularly, I don't know why yet.

Here is my personal cheat sheet for Transmission Daemon:

- Check if multiple instancesof Transmission Daemon are running:
`ps aux | grep transm`
there should be only one, run by the user in config (debian-transmission or custom-user, NOT ROOT)

- If necessary, kill them with:
`kill [PID]`

- Reload the service:
`systemctl start transmission-daemon` or `service transmission-daemon start`

- Place to check for config:
`/lib/systemd/system/transmission-daemon.service` (no idea, but this is the place where the user was changed back from custom user to debian-transmission without explanation, so the user should be double check here first)
`/etc/init.d/transmission-daemon` (seems to be another daemon config)
`/etc/init/transmission-daemon.conf `(system config for automatic launch of the service? also change the user running the service here)
`/etc/transmission-daemon/` (where the JSON user config stands)
`/var/lib/transmission-daemon/` (the user home folder of the debian-transmission user. The `/var/lib/transmission-daemon/.config/transmission-daemon/setting.json` file should point to `/etc/transmission-daemon/settings.json`. If transmission is run by Apache user for example, there should be a config folder in `/var/www/.config`, with its setting file pointing to `/etc/transmission-daemon/`)
Also:
`/root/.config/transmission-daemon/ `(this folder shouldn't exist! if it does, it PROBABLY means transmission-daemon is run as root, which is not what we want, and it means there a config error somewhere)

I wonder if we can create a GH issue about all those config places.

**Comment by kushwavez** [comment]
hmm I don't know about that unfortunately, after changing the settings I stopped/started the daemon several times, but it is remained fine with my user settings, even after a reboot. 

But I have to mention that I did a clean install after a lots of messing with transmission files, first `sudo apt remove transmission*`
then `cd /` `sudo find . -name "transmission*" -exec rm -r {} \;` and then installed again and configured the way you mentioned. No problems so far.

And about config and all this stuff, it would be really nice to have a "First run setup", so we don't have to waste a lots of time with these (simple!!!) things... But y'know Linux has to be hard & not user friendly, it's the rule (haha)
